python/lsst/ctrl/execute/slurmPlugin.py

```diff
@@ -188,6 +187,180 @@ def createAllocationFile(self, input):
+            # the current user (Owner) and are Idle vanilla universe jobs.
+            full_constraint = f"{owner}{jstat}{bps1}{bps2}{juniv}"
+            if verbose:
+                print("Find Large BPS Jobs:")
+                print(f"full_constraint {full_constraint}")
+            condorq_data = condor_q(
+                constraint=full_constraint,
+                schedds={schedd_name: scheddref},
+                projection=projection,
+            )
+
+            # Dictionaries of results need to be disassembled
+
+            if len(condorq_data) == 0:
+                print("No Large BPS Jobs.")
+
+            if len(condorq_data) > 0:
+                #
+                # Gather the Large jobs
+                #
+                condorq_bps_large = {}
+                if verbose:
+                    print("Large jobs exist, have been queried/fetched.")
+                condorq_bps = condorq_data[schedd_name]
+                for jid in list(condorq_bps.keys()):
+                    job = condorq_bps[jid]
+                    thisCores = job["RequestCpus"]
+                    thisMemory = job["RequestMemory"]
+                    if thisCores > autoCPUs or thisMemory > autoCPUs * memoryPerCore:
+                        condorq_bps_large[jid] = job
+
+                #
+                # Collect a list of the labels
+                #
+                job_labels = []
+                if verbose:
+                    print("Loop over list of Large Jobs")
+                for jid in list(condorq_bps_large.keys()):
+                    ajob = condorq_bps_large[jid]
+                    if verbose:
+                        print(jid)
+                        print(ajob["bps_job_label"])
+                    job_labels.append(ajob["bps_job_label"])
+
+                #
+                # Get a list of the unique labels
+                #
+                unique_labels = set(job_labels)
+
+                #
+                # Make a jobs dictionary with the unique labels as keys
+                #
+                if verbose:
+                    print("Loop over unique job label list")
+                label_dict = {}
+                for job_label in unique_labels:
+                    hash = hashlib.sha1(job_label.encode("UTF-8")).hexdigest()
+                    shash = hash[:6]
+                    empty_list = []
+                    label_dict[job_label] = empty_list
+
+                # Loop over the Large jobs and categorize
+                for jid in list(condorq_bps_large.keys()):
+                    ajob = condorq_bps_large[jid]
+                    this_label = ajob["bps_job_label"]
+                    this_list = label_dict[this_label]
+                    this_list.append(ajob)
+
+                for job_label in unique_labels:
+                    if verbose:
+                        print(f"\n{job_label}")
+                    existingGlideinsIdle = 0
+                    numberOfGlideinsRed = 0
```

Comment:
I have to dig through code every time to remind myself what this variable means.  Please add a comment or consider picking a different name.

Reply:
The "Red" refers to Reduced.     It is 
numberOfGlideinsRed = numberOfGlideins - existingGlideinsIdle
That is, you see that you need   numberOfGlideins=4   say 128 GB glideins for 4 such jobs, 
however you detect that  existingGlideinsIdle=1 is already in the Slurm queue,  idle. 
So you do not need to submit numberOfGlideins4, 
but rather  numberOfGlideinsRed=4 -1 -> 3 . 
So maybe "Reduced".

Reply:
I chnaged it to numberOfGlideinsReduced .